tests/e2e/test_internal_trades.py

```diff
@@ -1,66 +1,113 @@
+from __future__ import annotations
+
 import unittest
+from dataclasses import dataclass
 from datetime import datetime
+from enum import Enum
 
 from src.dune_analytics import DuneAnalytics, QueryParameter
 from src.fetch.period_slippage import add_token_list_table_to_query
-from src.file_io import File
-from src.models import Address, InternalTokenTransfer, Network
+from src.models import AccountingPeriod, Address, Network
+
+
+class TransferType(Enum):
+    """
+    Classifications of Internal Token Transfers
+    """
+
+    IN_AMM = "IN_AMM"
+    OUT_AMM = "OUT_AMM"
+    IN_USER = "IN_USER"
+    OUT_USER = "OUT_USER"
+    INTERNAL_TRADE = "INTERNAL_TRADE"
+
+    @classmethod
+    def from_str(cls, type_str: str) -> TransferType:
+        """Constructs Enum variant from string (case-insensitive)"""
+        try:
+            return cls[type_str.upper()]
+        except KeyError as err:
+            raise ValueError(f"No TransferType {type_str}!") from err
+
+
+@dataclass
+class InternalTransfer:
+    """Total amount reimbursed for accounting period"""
+
+    transfer_type: TransferType
+    token: Address
+    amount: int
+
+    @classmethod
+    def from_dict(cls, obj: dict[str, str]) -> InternalTransfer:
+        """Converts Dune data dict to object with types"""
+        return cls(
+            transfer_type=TransferType.from_str(obj["transfer_type"]),
+            token=Address(obj["token"]),
+            amount=int(obj["amount"]),
+        )
+
+    @staticmethod
+    def filter_by_type(
+        recs: list[InternalTransfer], transfer_type: TransferType
+    ) -> list[InternalTransfer]:
+        """Filters list of records returning only those with indicated TransferType"""
+        return list(filter(lambda r: r.transfer_type == transfer_type, recs))
+
+    @classmethod
+    def internal_trades(cls, recs: list[InternalTransfer]) -> list[InternalTransfer]:
+        """Filters records returning only Internal Trade types."""
+        return cls.filter_by_type(recs, TransferType.INTERNAL_TRADE)
 
 
 def token_slippage(
     token_str: str,
-    internal_trade_list: list[InternalTokenTransfer],
+    internal_trade_list: list[InternalTransfer],
 ) -> int:
     return sum(a.amount for a in internal_trade_list if a.token == Address(token_str))
 
 
-def get_internal_transfers(
-    dune: DuneAnalytics, tx_hash: str, period_start: datetime, period_end: datetime
-) -> list[InternalTokenTransfer]:
```

Comment:
This method was moved inside the test class which has most of these input parameters are attributes. Now we can call this method as `self.get_internal_transfers` passing in only the `tx_hash`